### PR TITLE
Wrap assumption violations in ErrorCollector

### PR DESCRIPTION
**Pull request by kcooney**
addError() and checkSucceeds() now wrap assumption violations.

### CODE REVIEW

File: src/main/java/org/junit/rules/ErrorCollector.java [hunk 1]
```diff
@@ -45,6 +46,14 @@ protected void verify() throws Throwable {
      * Adds a Throwable to the table.  Execution continues, but the test will fail at the end.
      */
     public void addError(Throwable error) {
+        if (error == null) {
+            throw new NullPointerException("Error cannot be null");
+        }
+        if (error instanceof AssumptionViolatedException) {
+            AssertionError e = new AssertionError(error.getMessage());
+            e.initCause(error);
+            error = e;
```
Comment (marcphilipp) [inline src/main/java/org/junit/rules/ErrorCollector.java]: Minor thing: I'm generally not happy with reassigning method parameters.

**Comment by kcooney** [comment]
Note there are two commits, and I want to keep them separate. I will be happy to do the merge.

If this is approved I'll also merge #1371

**Comment by kcooney** [comment]
Merged.

**Comment by kcooney** [comment]
added to the release notes

**Pull request closed without merging**